jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/RetrieveRegistryCredentialsStep.java

```diff
@@ -178,10 +195,30 @@ public Authorization call() throws IOException, NonexistentDockerCredentialHelpe
        * public and does not need extra credentials) and return null.
        */
       buildLogger.info("No credentials could be retrieved for registry " + registry);
+
       return null;
     }
   }
 
+  @VisibleForTesting
+  static boolean isDockerHubRegistry(String registry) {
+    return DOCKER_HUB_REGISTRIES.stream().anyMatch(registry::equals);
+  }
+
+  @VisibleForTesting
+  Authorization findOtherDockerHubCredential(String dockerHubRegistry)
```

Comment:
Generally, we want to try to keep the build steps (`AsyncStep`) as minimal as possible and delegate the real logic to other classes. So, here for example, we should have probably put the credential helper inference logic inside a `DockerCredentialHelperInferer` rather than in this `AsyncStep` so that we could unit test against it well.

Reply:
Just to verify, what I understand from your comments is to minimizing code lines in AsyncStep classes by moving code out to different classes, and not about minimizing operations and time the steps take, right?